src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -816,7 +818,18 @@ grpc_chttp2_stream* grpc_chttp2_parsing_accept_stream(grpc_chttp2_transport* t,
   if (t->channel_callback.accept_stream == nullptr) {
     return nullptr;
   }
-  grpc_chttp2_stream* accepting;
+  if (t->resource_user != nullptr &&
+      !grpc_resource_user_safe_alloc(t->resource_user,
+                                     GRPC_RESOURCE_QUOTA_CALL_SIZE)) {
+    gpr_log(GPR_DEBUG, "Memory exhausted, rejecting the stream.");
+    grpc_slice_buffer_add(
```

Comment:
yeah, this is exactly what I wanted. Can you please add a comment here to explain why and what we are doing here?